src/main/java/com/example/springboot/service/UserService.java

```diff
@@ -0,0 +1,158 @@
+            throw new InvalidException(ResultInfoConstants.NUMBER_DOES_NOT_EXIST);
+        }
+        Long otp = ThreadLocalRandom.current().nextLong(1000, 1000000);
+        log.info("The Otp generated is: {}", otp);
+        oneTimePassword.setPassword(otp);
+        oneTimePassword.setMobile(mobileNumber);
+        return otp;
+    }
+
+    public Long validOtp(Long otp) {
+        Long generatedOtp = oneTimePassword.getPassword();
+        Long enteredOtp = otp;
+        if (!generatedOtp.equals(enteredOtp)) {
+            throw new InvalidException(ResultInfoConstants.INVALID_OTP);
+        }
+        return oneTimePassword.getMobile();
+    }
+
+
+    public Long login(@Valid OldUser user) {
+        Long mobileNumber = user.getUsername();
+        Long mPin = user.getPassword();
+        String mobile = mobileNumber.toString();
+        if (mobile.length() != 10) {
+            throw new InvalidException(ResultInfoConstants.INVALID_MOBILE);
+        }
+        List<Long> allMobiles = userRepository.getAllMobileNumbers();
+        if (!allMobiles.contains(mobileNumber)) {
+            throw new InvalidException(ResultInfoConstants.NUMBER_DOES_NOT_EXIST);
+        }
+        Long mpin = userRepository.getMpin(mobileNumber);
+        if (!mpin.equals(mPin)) {
+            throw new InvalidException(ResultInfoConstants.INVALID_MPIN);
+        }
+        UserTable userTable = userRepository.getUser(mobileNumber);
+        userRepository.save(userTable);
+        return mobileNumber;
+    }
+
+    public Long signout(Long mobile) {
+        String mobileNumber = mobile.toString();
+        if (mobileNumber.length() != 10) {
+            throw new InvalidException(ResultInfoConstants.INVALID_MOBILE);
+        }
+        List<Long> allMobiles = userRepository.getAllMobileNumbers();
+        if (!allMobiles.contains(mobile)) {
+            throw new InvalidException(ResultInfoConstants.USER_NOT_FOUND);
+        }
+        UserTable userTable = userRepository.getUser(mobile);
+        userRepository.save(userTable);
+        return mobile;
+    }
+
+//    public Long delete(Long mobile) {
+//        String mobileNumber = mobile.toString();
+//        if (mobileNumber.length() != 10) {
+//            throw new InvalidNumberException(ResultInfoConstants.INVALID_MOBILE);
+//        }
+//        List<Long> allMobiles = userRepository.getAllMobileNumbers();
+//        if (!allMobiles.contains(mobile)) {
+//            throw new InvalidNumberException(ResultInfoConstants.USER_NOT_FOUND);
+//        }
+//       userRepository.deleteAccount(mobile);
+//        return mobile;
+//    }
+
+    public Long forgotPassword(Long mobile) {
+        String mobileNumber = mobile.toString();
+        if (mobileNumber.length() != 10) {
+            throw new InvalidException(ResultInfoConstants.INVALID_MOBILE);
+        }
+        List<Long> allMobiles = userRepository.getAllMobileNumbers();
+        if (!allMobiles.contains(mobile)) {
```

Comment:
Incorrect. You need to query database with where clause and see if it returns any values.